gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FeatureDocument.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.remote.firestore.schema;
+
+import androidx.annotation.Nullable;
+import com.google.android.gnd.persistence.remote.firestore.AuditInfoDoc;
+import com.google.firebase.firestore.GeoPoint;
+import com.google.firebase.firestore.IgnoreExtraProperties;
+
+/** Feature entity stored in Firestore. */
+@IgnoreExtraProperties
+class FeatureDocument {
+  @Nullable private String featureTypeId;
+  @Nullable private String customId;
+  @Nullable private String caption;
+  // TODO: Replace with consistent name throughout.
+  @Nullable private GeoPoint center;
+  @Nullable private AuditInfoDoc created;
+  @Nullable private AuditInfoDoc modified;
+
+  FeatureDocument(
+      @Nullable String featureTypeId,
+      @Nullable String customId,
```

Comment:
What exactly is customId as opposed to Id? Is this set by the user? Does it have to be unique?

Reply:
Yes, this is a reserved field to allow the user to upload a custom id in the CSV. We'll need this for preloaded data.